x-pack/plugins/security_solution/common/detection_engine/rule_schema/model/specific_attributes/query_attributes.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { LimitedSizeArray } from '@kbn/securitysolution-io-ts-types';
+
+export const AlertSuppressionGroupBy = LimitedSizeArray({
+  codec: t.string,
+  minSize: 1,
+  maxSize: 3,
+});
+
+/**
+ * Schema for fields relating to alert suppression, which enables limiting the number of alerts per entity.
+ * e.g. group_by: ['host.name'] would create only one alert per value of host.name. The created alert
+ * contains metadata about how many other candidate alerts with the same host.name value were suppressed.
+ */
+export type AlertSuppression = t.TypeOf<typeof AlertSuppression>;
+export const AlertSuppression = t.exact(
+  t.type({
+    group_by: AlertSuppressionGroupBy,
+  })
+);
+
+export const minimumLicenseForSuppression = 'platinum';
```

Comment:
I switched from validating the params on create/edit of a rule to checking the license at [runtime](https://github.com/elastic/kibana/pull/142686/files#diff-83cd1a1d3828f621d8c48c80f372480c602727d1c58e9217d3bbb2a027ce7d56R66) in the executor to avoid edge cases around license expiration/degradation. E.g. if the license expires, rules would be exportable but then would fail to import, `patch` would allow editing a rule while leaving suppression params untouched but `update` would not, etc.

The behavior now is that if a license is insufficient and suppression is enabled then rules will continue to work, but the suppression configuration will be ignored. If the platinum license is restored, then suppression will start to work again on those rules automatically. This way we can keep the functionality behind the license appropriately but not cause issues with rule management if the license level does drop.

Reply:
Thanks for explaining, that makes sense 👍 

> I switched from validating the params on create/edit of a rule to checking the license at [runtime](https://github.com/elastic/kibana/pull/142686/files#diff-83cd1a1d3828f621d8c48c80f372480c602727d1c58e9217d3bbb2a027ce7d56R66) in the executor to avoid edge cases around license expiration/degradation

Btw, can we use there `minimumLicenseForSuppression` instead of hardcoded `platinum` value?

Reply:
Oops, yeah that should be `minimumLicenseForSuppression` as well. I'll add it to the follow up list.